### ISSUE TITLE
Re-organize Model packages, ScalaDoc

### ISSUE THREAD

**Issue by maldins46** [scalaquest/PPS-19-ScalaQuest#35]
As a final effort to define the model's basic structure, we are going to re-organize the model packages, in a way that makes clearer the behavioral implementation. In addition, we are going to finish the ScalaDoc in our part (model, interpreter, reducer, resolver).

**Comment by maldins46** [comment]
This morning we worked on a heavy refactoring of the terms used into the model, as stated in 2a1877ee58ae218ca2dd863881d684fb423659b5, and finished the `Model` ScalaDoc. In particular:
- Keyword **behaviorable** has been changed to **BehaviorBased**. This is a big change as involves a lot of classes.
- Modified the package structure. Now, `behaviorBased` is the only child package of `Model`, containing all the behavior-based-fashioned implementations. The package presents `common` as a child, at the same level of a `impl` child, that contains a standard implementation that makes use of `common`.
- `impl` packages now contain classes named `Simple*`, as it is a common pattern across the project. This can be discussed, and maybe some other keywords could be used;
- Other minor refactoring to maintain coherence between terms such as `Triggers` becoming `ItemTriggers`, `Behavior` becoming `ItemBehavior`, equivalent refactoring for  `Ground`;
- `ItemRef` has been split from the `ItemRetriver`. `ItemRef` something that belongs to the `Model`, meanwhile the `ItemRetriver` is a concept used only inside the `Interpreter`. Then itemRetriver has been renamed to `RefToItem`.
- `Direction` enum is now included into the `Room` companion object, as it is a proper concept of the `Room`. Directions are not an enum anymore, but case classes of a trait `Direction`, making eventually possible to the storyteller to introduce their proper directions.

In light of these changes, we should also update the Ubiquitous language.
